irc_commands.c

```diff
@@ -57,6 +58,117 @@ static void irc_cmd_pass(irc_t *irc, char **cmd)
 	}
 }
 
+static gboolean irc_sasl_plain_parse(char *input, char **user, char **pass)
+{
+	int len;
+	int i = 0;
+	int part = 0;
+	guint8 *decoded;
+	char *parts[2];
+
+	/* bitlbee's base64_decode wrapper adds an extra null terminator at the end */
+	len = base64_decode(input, &decoded);
+
+	while (i < len && part < 3) {
```

Comment:
I think `for (i = 0, part = 0; i < len && part < 3; part++)` could work. Maybe with some variable renaming